### ISSUE TITLE
Issue testing KAS-IFC with val + responder tests

### ISSUE THREAD

**Issue by abkarcher** [usnistgov/ACVP-Server#217]
**environment**
Demo

**testSessionId**
259262

Hello,

I have been trying to debug a harness implementation of KAS-IFC, specifically an issue where some VAL tests with a "responder" role were failing with the reason "FailChangedZ". Libacvp is detecting that the tests passed.

I think I may have isolated a possible cause, related to (pending) changes I made after https://github.com/usnistgov/ACVP/issues/1338.

After that conversation, I have changed the library side of libacvp to make comparisons against hashZ when a hash function is provided, and ignore the provided Z value; the Z value seemed to be redundant and caused extra complexity in the client.

However, with this particular failure reaason "FailChangedZ", my idea is that the server is altering the Z value to cause a failure in the VAL test, but not altering the hashZ value. Thus, I am not able to pass the test properly. If this is the case, could the behavior be modified so the hashZ value also gets modified?

I could be wrong; I have been looking at this code for a bit too long now :) Any help isolating this issue would be greatly appreciated!

Thanks,
Andrew

**Comment by livebe01** [comment]
Hi @abkarcher, 
At first look, I think you might be right on this. But let me dig deeper to verify that I'm understanding correctly what I'm looking at.  
Thanks,
 Ben

**Comment by abkarcher** [comment]
Hi,

Is there any update on this? I only ask because it will require me to redo some libacvp code in the coming month or two if the raw Z value is required.

Thanks,
Andrew

**Comment by livebe01** [comment]
Yes! We plan to have a fix for this in the next week or two. We'd like to send a new release to Demo next week if possible. Sorry for the wait on this.

**Comment by abkarcher** [comment]
Thanks for the update 😄 

Andrew

**Comment by livebe01** [comment]
This change is on Demo in release [v1.1.0.26](https://github.com/usnistgov/ACVP-Server/releases/tag/v1.1.0.26)

**Comment by livebe01** [comment]
This change is on Prod in release [v1.1.0.26](https://github.com/usnistgov/ACVP-Server/releases/tag/v1.1.0.26)
